selftests/functional/test_sysinfo.py

```diff
@@ -49,6 +54,50 @@ def test_sysinfo_enabled(self):
             msg = f"The sysinfo/{hook} subdirectory does not exist:\n{result}"
             self.assertTrue(os.path.exists(sysinfo_subdir), msg)
 
+    def test_sysinfo_per_test_enabled(self):
+        config_path = os.path.join(self.tmpdir.name, "config.conf")
+        script.make_script(config_path, PER_TEST_CONF)
+        cmd_line = (
+            f"{AVOCADO} --config {config_path} run --job-results-dir {self.tmpdir.name} "
+            f"examples/tests/passtest.py"
+        )
+        result = process.run(cmd_line)
+        expected_rc = exit_codes.AVOCADO_ALL_OK
+        self.assertEqual(
+            result.exit_status,
+            expected_rc,
+            f"Avocado did not return rc {expected_rc}:\n{result}",
+        )
+        test_results_path = os.path.join(self.tmpdir.name, "latest", "test-results")
+        sysinfo_dirs = [
+            sysinfodir
+            for sysinfodir in os.listdir(test_results_path)
+            if "-sysinfo-" in sysinfodir
+        ]
+        self.assertEqual(
+            len(sysinfo_dirs), 2, "Avocado didn't create sysinfo directories"
+        )
+
+    def test_sysinfoi_per_test_disabled(self):
+        cmd_line = (
+            f"{AVOCADO} run --job-results-dir {self.tmpdir.name} "
+            f"examples/tests/passtest.py"
+        )
+        result = process.run(cmd_line)
+        expected_rc = exit_codes.AVOCADO_ALL_OK
+        self.assertEqual(
+            result.exit_status,
+            expected_rc,
+            f"Avocado did not return rc {expected_rc}:\n{result}",
+        )
+        test_results_path = os.path.join(self.tmpdir.name, "latest", "test-results")
+        sysinfo_dirs = [
+            sysinfodir
+            for sysinfodir in os.listdir(test_results_path)
+            if "-sysinfo-" in sysinfodir
```

Comment:
In previous Avocado versions, the per-test sysinfo would be placed at:

 * test-results/$test_id/sysinfo/pre
 * test-results/$test_id/sysinfo/post

I understand this is simply following the default task behavior, but I think usability wise this is worse.  Have you thought about implementing the earlier behavior described above?

Reply:
> Yes, I was thinking about that. The problem is that each task has separated directory and for now we don't have any way how the plugin could change that so I have to use this approach instead. Maybe I can remove sysinfo from this PR and solve this problem before I will introduce sysinfo or we will go with this solution and solve it later.

Yes, I agree with the removal of `sysinfo` from this PR simply because I think it's better to present users with the `sysinfo` functionality working as expected/documented.